crates/daemon/src/wallet.rs

```diff
@@ -512,6 +591,7 @@ mod tests {
                     time_to_lock,
                 },
                 blockchain_client: (),
+                db: None,
```

Comment:
❓ Why is this field an `Option`? I would expect to always be able to provide the database.

Reply:
@luckysori well, I don't like it either, but I didn't find a better way of doing it with the existing tests. Since the tests mainly use a MemoryDatabase, I do not have a Db available there. Do you have a suggestion?

Reply:
Hmm, I didn't come up with anything after some limited experimenting. I might come back to this later myself.